### ISSUE TITLE
Some questions about training

### ISSUE THREAD

**Issue by sunjc0306** [YuliangXiu/ICON#42]
I would like to know some details about training. 
Is the Ground-Truth SMPL  or the predicted SMPL used in training ICON?
Also, what about normal images?
According to my understanding of the paper and practice, ICON should train the normal network first and then train the implicit reconstruction network. 
When I reproduce ICON, I don't know whether to choose the Ground-Truth or the predicted data for SMPL model and normal images, respectively.

**Comment by YuliangXiu** [comment]
GT SMPL is used for training, which is the same case for normal network training.

Actually, the normal and implicit networks can be trained separately and simultaneously, I used GT normal images to train the normal network.

**Comment by sunjc0306** [comment]
Thanks for your reply.  I maybe know some details.
"I used GT rendered normal images to train the normal network instead of the predicted normal images."---I don't quite understand this sentence.
GT rendered normal images are surely used for training the normal network since they serve as labels to supervise training.
However, If the normal and implicit networks are trained separately, which image is chosen during the training of the implicit networks, GT rendered normal images or predicted normal images from the normal network?

In addition, I would like to ask a few questions about testing and evaluation.
I have tested in THuman2.0 with your official version and these results reconstructed from ICON are very great in qualitative evaluation. For example, fingers and face details are even obtained. However, their quantitative metrics（including CD PSD l2 COS）are terrible. I would like to know if you encounter such a case that qualitative results outperform but quantitative metrics are terrible?
![image](https://user-images.githubusercontent.com/34236842/162657882-6adb1c87-bab9-4202-8ab3-a1bd437c4857.png)
![image](https://user-images.githubusercontent.com/34236842/162657968-1c699d16-3664-4471-a353-259cf10d9749.png)

**Comment by sunjc0306** [comment]
These doubts are Crucial issues when I train and test ICON. Thanks for your help again, which will better help me to learn and reproduce ICON.

**Comment by YuliangXiu** [comment]
Now both `dataset preprocess code` and `training code` are ready!

[Dataset processing](https://github.com/YuliangXiu/ICON/blob/master/docs/dataset.md)
[Training](https://github.com/YuliangXiu/ICON/blob/master/docs/training.md)

**Comment by sunjc0306** [comment]
Thanks for your reminder. 
I have implemented retraining on the DeepHuman dataset and THuman2.0 dataset using the SMPL model and semantic map.
It finds that the features of ICON are extremely efficient, especially SMPL features, so that excellent models can be reconstructed with so few features. Your work has also inspired me how to extract pixel features and spatial features efficiently.
Thank you again.

**Comment by YuliangXiu** [comment]
@sunjc0306 New cloth-refinement module is released. Use `-loop_cloth 200` to refine ICON's reconstruction, making it as good as the predicted clothing normal image.
![overlap](https://user-images.githubusercontent.com/7944350/181116211-2ed4ab54-73b1-487d-9496-15019dfdbc63.gif)
